### PR TITLE
NavigationOptions refactoring

### PR DESCRIPTION
**Pull request by tmat**
Validate builds:
- [x] TypeScript
- [x] F#

### CODE REVIEW

File: src/Tools/ExternalAccess/FSharp/Navigation/FSharpDocumentNavigationService.cs [hunk 1]
```diff
@@ -50,31 +53,34 @@ public bool CanNavigateToPosition(Workspace workspace, DocumentId documentId, in
...
         public bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, OptionSet options)
-            => TryNavigateToSpan(workspace, documentId, textSpan, options, CancellationToken.None);
+            => TryNavigateToSpan(workspace, documentId, textSpan, CancellationToken.None);
 
-        public bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, OptionSet options, CancellationToken cancellationToken)
+        public bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, CancellationToken cancellationToken)
         {
             var service = workspace.Services.GetService<IDocumentNavigationService>();
-            return service.TryNavigateToSpan(workspace, documentId, textSpan, options, cancellationToken);
+            return service.TryNavigateToSpan(workspace, documentId, textSpan, NavigationOptions.Default with { PreferProvisionalTab = true }, cancellationToken);
```
Comment (JoeRobich) [inline src/Tools/ExternalAccess/FSharp/Navigation/FSharpDocumentNavigationService.cs:63]: Why are we setting `PreferProvisionalTab` to `true` for F#? Are they currently using the OptionSet to control this behavior?

Comment (tmat) [reply src/Tools/ExternalAccess/FSharp/Navigation/FSharpDocumentNavigationService.cs:63]: Yes, they set it to true.

File: src/Features/Core/Portable/DocumentSpanExtensions.cs [hunk 2]
```diff
@@ -14,48 +14,24 @@ namespace Microsoft.CodeAnalysis
 {
     internal static class DocumentSpanExtensions
     {
-        [Obsolete("Use CanNavigateToAsync instead", error: false)]
-        public static bool CanNavigateTo(this DocumentSpan documentSpan, CancellationToken cancellationToken)
-        {
-            var workspace = documentSpan.Document.Project.Solution.Workspace;
-            var service = workspace.Services.GetService<IDocumentNavigationService>();
-            return service.CanNavigateToSpan(workspace, documentSpan.Document.Id, documentSpan.SourceSpan, cancellationToken);
-        }
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/DocumentSpanExtensions.cs:23]: nice.  no concern about our partners?

File: src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptDefinitionItemBase.cs [hunk 3]
```diff
@@ -2,11 +2,16 @@
...
+using System;
 using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Navigation;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
 {
+    [Obsolete]
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptDefinitionItemBase.cs:14]: is TS appraised of the steps they'll need to take?

**Review by CyrusNajmabadi — APPROVED** [review]
lgtm as long as partners are looped into how respond gently on their end.

**Comment by tmat** [comment]
Yes, TS and F# are being handled.

**Pull request merged**